core-109.60.00 fails to build on OpenBSD 5.4

As a workaround, I added `|| defined(__OpenBSD__)` to the macro conditional on line 1565 of `lib/unix_stubs.c` (since I don't need multicast functionality right now) and compiling succeeded.

```
$ gmake
./setup.exe -build 
W: Cannot find source file matching module 'core' in library core
Finished, 0 targets (0 cached) in 00:00:00.
+ ocamlfind ocamlc -I lib -package core_kernel -c lib/unix_stubs.c
lib/unix_stubs.c:680:2: warning: #warning "_POSIX_SYNCHRONIZED_IO undefined or <= 0; aliasing unix_fda
tasync to unix_fsync"
lib/unix_stubs.c:952:2: warning: #warning "posix timers not present; clock functions undefined"
lib/unix_stubs.c: In function 'core_unix_mcast_modify':
lib/unix_stubs.c:1568: error: storage size of 'mreq_source' isn't known
lib/unix_stubs.c:1577: error: 'IP_ADD_SOURCE_MEMBERSHIP' undeclared (first use in this function)
lib/unix_stubs.c:1577: error: (Each undeclared identifier is reported only once
lib/unix_stubs.c:1577: error: for each function it appears in.)
lib/unix_stubs.c:1695:2: warning: #warning "_POSIX_PRIORITY_SCHEDULING not present; sched_setscheduler
 undefined"
Command exited with code 2.
Compilation unsuccessful after building 23 targets (22 cached) in 00:00:00.
E: Failure("Command ''/usr/local/bin/ocamlbuild' lib/libcore_stubs.a lib/dllcore_stubs.so lib/core.cma
 lib/core.cmxa lib/core.a lib/core.cmxs top/core_top.cma top/core_top.cmxa top/core_top.a top/core_top
.cmxs -use-ocamlfind -tag debug' terminated with error code 10")
gmake: *** [build] Error 1
```

--- a/lib/unix_stubs.c
+++ b/lib/unix_stubs.c
@@ -1562,7 +1562,7 @@ CAMLprim value core_unix_mcast_modify (value v_action,
       }
 
       if (Is_block(v_source_opt)) {
-#if defined(__APPLE__)
+#if defined(__APPLE__) || defined(__OpenBSD__)
         caml_failwith("core_unix_mcast_modify: ~source is not supported on MacOS");
 #else
         struct ip_mreq_source mreq_source;

